### PR TITLE
Add check for non-ascii fix:#4268

### PR DESCRIPTION
**Pull request by rhuanjl**
This PR adds an additional test to CI to block non-ascii characters, the tests, tools and Build folders are excluded from the check.

This is based on the eol check test but with a different grep command.

Additionally I found some non-ascii at the start of JavascriptString.cpp so have removed it.

@dilijev please can you take a look

### CODE REVIEW

**Comment by rhuanjl** [comment]
@dotnet-bot test ci please

File: jenkins/check_file_ascii.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#-------------------------------------------------------------------------------------------------------
+
+ERRFILE=check_ascii.sh.err
+ERRFILETEMP=$ERRFILE.0
+
+# display a helpful message for someone reading the log
+echo "Check ascii > Checking $1"
+
+if [ ! -e $1 ]; then # the file wasn't present; not necessarily an error
+    echo "WARNING: file not found: $1"
```
Comment (MSLaguana) [inline jenkins/check_file_ascii.sh]: Is this to account for the case of deleted files?

File: lib/Runtime/Library/JavascriptString.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿//-------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------
```
Comment (MSLaguana) [inline lib/Runtime/Library/JavascriptString.cpp]: What happened here? Was there a byte order marker?

**Comment by MSLaguana** [comment]
@dotnet-bot test ci please

File: lib/Runtime/Library/JavascriptString.cpp (hunk 2 above)
Comment (rhuanjl) [reply lib/Runtime/Library/JavascriptString.cpp]: Yes - oddly this was the only file in the source tree with one.

**Comment by MSLaguana** [comment]
Seems to work ok: https://ci2.dot.net/job/Microsoft_ChakraCore/job/master-ci/job/GenPRTest/job/ubuntu_check_ascii_prtest/1/console

**Comment by MSLaguana** [comment]
Oh, spoke too soon:
`grep: Invalid range end`

**Comment by rhuanjl** [comment]
@MSLaguana thank you for the review and sorry for the error - it had run fine offline.

**Comment by MSLaguana** [comment]
@dotnet-bot test ci please

File: jenkins/check_file_ascii.sh (hunk 1 above)
Comment (dilijev) [reply jenkins/check_file_ascii.sh]: Yep. I believe this logic was copies from another one of our checks.

**Pull request merged**